### PR TITLE
feat: signup form

### LINKED ISSUE #90: feat: Signup form
**Is your feature request related to a problem? Please describe.**
When loading the website for the first time, it is important for the user to be able to register. The signup form should distinguish between company users, and regular users.

**Describe the solution you'd like**
A register/signup form that should include the following input fields. All should be required

All user types:
- [ ] E-mail
- [ ] First name 
- [ ] Surname
- [ ] Password
- [ ] Password confirmation

Regular user:
- [ ] Study program (Option field)
- [ ] Current study year (Option field)

Company user:
- [ ] Registration code (auth code provided by Auth administration)

### PR DESCRIPTION
**Pull request by ipeglin**
# Changes
* New Signup component for registering users
* Signup button that registeres a new user in the database
* New SelectInput component for selecting study programs and current year
* New TextInput component for providing name in form

# Might change
* Remove Select and Text input components in favour of using the vuetify components directly in the form component.

# Remaining work
* Input `<UserEmailinput />` and `<UserPasswordInput />` components (#78 and #77) in register form

### CODE REVIEW

File: components/user/SignupForm.vue [hunk 1]
```diff
@@ -0,0 +1,100 @@
+<script setup lang="ts">
+  const formInput = ref({
+    name: null,
+    email: null,
+    password: null,
+    studyProgram: null,
+    currentYear: null,
+    userType: 'student',
```
[the remark this answers is not in the document]
Comment (ipeglin) [reply components/user/SignupForm.vue:8]: I'm setting the default userType to undefined as of now. I can change it later somewhere down the road if it is needed

File: components/user/SignupForm.vue [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      password: '123456',
+      studyProgram: 'Elektronisk systemdesign og innovasjon',
+      name: 'Test User',
+    })
+  }
+</script>
+
+<template>
+  <v-form>
+    <v-container>
```
[the remark this answers is not in the document]
Comment (ipeglin) [reply components/user/SignupForm.vue]: I'm going to work on this during the workshop on tuesday

File: components/user/TextInput.vue
[the remark this answers is not in the document]
Comment (ipeglin) [reply components/user/TextInput.vue:1]: I agree. This draft is only a work-in-progress, so it is by no means finished :) I'll have a look at it tomorrow

**Comment by AasmundN** [comment]
Need to wait for #102 too. I'll have it done before next meeting.

File: components/user/SignupForm.vue [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    if (!form.value) return
+    const { valid } = await form.value.validate()
+
+    if (!valid) throw new Error('Form is not valid')
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { passwordConfirmation, userType, ...user } = state
+
+    if (userType === 'student') {
+      await registerUser({
```
Comment (AasmundN) [inline components/user/SignupForm.vue:35]: The registerUser function will throw errors if something goes wrong. These must be handled with _trycatch_ blocks and proper feedback to the user through some sort of popup.

**Review by AasmundN — CHANGES_REQUESTED** [review]
I fixed some missing translations and added a max-width to the form. After merging #102 and fixing error handling of the `registerUser` function this should be good to go.

**Review by AasmundN — APPROVED** [review]
LGTM

**Pull request merged**